databricks_cli/workspace/api.py

```diff
@@ -0,0 +1,126 @@
+# limitations under the License.
+
+import os
+from base64 import b64encode, b64decode
+
+import click
+
+from databricks_cli.configure.config import get_workspace_client
+from databricks_cli.dbfs.exceptions import LocalFileExistsException
+
+
+DIRECTORY = 'DIRECTORY'
+NOTEBOOK = 'NOTEBOOK'
+LIBRARY = 'LIBRARY'
+
+
+class WorkspaceFileInfo(object):
+    def __init__(self, path, object_type, language=None):
+        self.path = path
+        self.object_type = object_type
+        self.language = language
+
+    def to_row(self, is_long_form, is_absolute):
+        path = self.path if is_absolute else self.basename
+        if self.is_dir:
+            stylized_path = click.style(path, 'cyan')
+        elif self.is_library:
+            stylized_path = click.style(path, 'green')
+        else:
+            stylized_path = path
+        if is_long_form:
+            return [self.object_type, stylized_path, self.language]
+        else:
+            return [stylized_path]
+
+    @property
+    def is_dir(self):
+        return self.object_type == DIRECTORY
+
+    @property
+    def is_notebook(self):
+        return self.object_type == NOTEBOOK
+
+    @property
+    def is_library(self):
+        return self.object_type == LIBRARY
+
+    @property
+    def basename(self):
+        return os.path.basename(self.path)
+
+    @classmethod
+    def from_json(cls, deserialized_json):
+        return cls(**deserialized_json)
+
+
+def get_status(workspace_path):
+    workspace_client = get_workspace_client()
+    return WorkspaceFileInfo.from_json(workspace_client.get_status(workspace_path))
+
+
+def list_objects(workspace_path):
+    workspace_client = get_workspace_client()
+    response = workspace_client.list(workspace_path)
+    # This case is necessary when we list an empty dir in the workspace.
+    # TODO(andrewmchen): We should make our API respond with a json with 'objects' field even
+    # in this case.
+    if 'objects' not in response:
+        return []
+    objects = response['objects']
+    return [WorkspaceFileInfo.from_json(f) for f in objects]
+
+
+def mkdirs(workspace_path):
+    workspace_client = get_workspace_client()
+    workspace_client.mkdirs(workspace_path)
+
+
+def import_workspace(source_path, target_path, language, fmt, is_overwrite):
+    workspace_client = get_workspace_client()
+    with open(source_path, 'r') as f:
+        content = b64encode(f.read())
+        workspace_client.import_workspace(
+            target_path,
+            fmt,
+            language,
+            content,
+            is_overwrite)
+
+
+def export_workspace(source_path, target_path, fmt, is_overwrite):
+    if os.path.exists(target_path) and not is_overwrite:
+        raise LocalFileExistsException()
+    workspace_client = get_workspace_client()
+    output = workspace_client.export_workspace(source_path, fmt)
+    content = output['content']
+    # Will truncate target_path.
```

Comment:
What do you mean truncate?

Reply:
I meant it will overwrite the target path. I can make it more clear.